Source/Manager.swift

```diff
@@ -9,144 +9,159 @@
+      pathComponents.removeFirst()
     }
-    
-    // MARK: - Helpers
-    
-    /// Normalize a path which can be used as a Resource Identifier and requested resource IDs, if available.
-    /// - Returns: "/path/to/resource/123?foo=bar&baz=quux" -> ("/path/to/resource/", 123, ["foo": ["bar"], "baz": ["quux"])
-    static func resourceArtifactsFromURL(url: NSURL) -> (name: NameArtifact, ids: IDArtifacts?, query: QueryArtifacts?)? {
-        guard var pathComponents = url.pathComponents else { return nil }
-
-        var name: String?
-        var ids = [Int]?()
-        let separator = "/"
-
-        // Check if the URL has an ID within it -> /api/path/to/123/example
-        for (index, component) in pathComponents.enumerate() {
-
-            // if it does...
-            if let id = Int(component) {
-
-                // add it to the IDs array to be returned
-                if ids?.append(id) == nil {
-                    ids = [id]
-                }
-                
-                // remove the number and replace with predetermined key to be used for the name to be returned
-                pathComponents[index] = ":id"
-            }
-        }
 
-        // prevents "//" from occurring when joining the path components
-        if pathComponents.first == separator {
-            pathComponents.removeFirst()
-        }
-        
-        name = pathComponents.joinWithSeparator(separator)
-        name = separator + name!
+    name = pathComponents.joinWithSeparator(separator)
+    name = separator + name!
 
-        let query = queryDictionaryFromURL(url)
-        
-        return (name!, ids, query)
+    let body = ArtifactDictionaryFromData(request.HTTPBody)
+    let query = ArtifactDictionaryFromString(url.query)
+
+    return (name!, ids, query, body)
+  }
+
+  /// Return a dictionary with arrays populated with the values of HTTP body data. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromData(data: NSData?) -> QueryArtifacts? {
+    guard let data = data,
+      let componentString = String(data: data, encoding: NSUTF8StringEncoding)
+      else {
+        return nil
     }
 
-    // inspired by https://gist.github.com/freerunnering/1215df277d750af71887
-    /// Return a dictionary with arrays populated with the values of query parameters. URLs allow keys to be declared multiple times, hence the array container.
-    static func queryDictionaryFromURL(url: NSURL) -> QueryArtifacts? {
-        guard let query = url.query else { return nil }
+    return ArtifactDictionaryFromString(componentString)
+  }
 
-        var queryDict = QueryArtifacts()
+  // inspired by https://gist.github.com/freerunnering/1215df277d750af71887
+  /// Return a dictionary with arrays populated with the values of query parameters. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromString(string: String?) -> QueryArtifacts? {
```

Comment:
"any func that can be private should be made private"
How do you feel about that guideline?